crates/wasi-common/yanix/src/sys/linux/mod.rs

```diff
@@ -8,8 +8,23 @@ use crate::dir::SeekLoc;
 use std::io::Result;
 
 impl SeekLoc {
+    #[cfg(target_pointer_width = "64")]
     pub unsafe fn from_raw(loc: i64) -> Result<Self> {
         let loc = loc.into();
         Ok(Self(loc))
     }
+    #[cfg(target_pointer_width = "32")]
```

Comment:
Is the `#[cfg]` here necessary? 

I'd love to have as few `#[cfg]` as necessary and it seems like we'd want this logic on all platforms?

Reply:
Agreed about `#[cfg]`s.

Hmm, do you reckon we could collapse both `from_raw` versions into one with `try_into`?

Reply:
Ah yeah @sunfishcode [put it well here](https://github.com/bytecodealliance/wasmtime/pull/1269#discussion_r390414227)